fs/xfstests.py

```diff
@@ -201,6 +201,12 @@ def setUp(self):
                 packages.remove('indent')
                 packages.remove('btrfs-progs-devel')
 
+            if self.detected_distro.name == 'rhel' and\
+                    self.detected_distro.version.startswith('9'):
+                packages.remove('dump')
+                packages.remove('indent')
+                packages.remove('btrfs-progs-devel')
+
```

Comment:
@disgoel can this be tweaked like,

packages_remove = ['dump', 'indent', 'btrfs-progs-devel'] 
packages = list(set(packages) - set(packages_remove))

So that, in future if need to remove any more packages, adding it to remove list will be easier task and maintains code reliability, instead of adding multiple packages.remove statements

And also, please attach the job.log file

Reply:
@Naresh-ibm please find the job log
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/7897253/job.log)

Reply:
@disgoel I could see some errors in the job.log, as below,

127 generic/231 generic/233 generic/247 generic/263 generic/317 generic/388 generic/459 generic/648 xfs/007 xfs/107 xfs/116 xfs/148 xfs/300 xfs/540 xfs/542
2022-01-19 18:16:51,013 process          L0470 DEBUG| [stdout] Failed 23 of 1034 tests
2022-01-19 18:16:51,016 process          L0470 DEBUG| [stdout] 
2022-01-19 18:16:51,022 process          L0739 INFO | Command './check  -g auto' finished with 1 after 7138.883506954s
2022-01-19 18:16:51,023 xfstests         L0366 INFO | ERR: Test(s) failed. Message: Test error. Failed 23 of 1034 tests.
2022-01-19 18:16:51,023 stacktrace       L0039 ERROR| 
2022-01-19 18:16:51,023 stacktrace       L0041 ERROR| Reproduced traceback from: /usr/local/lib/python3.9/site-packages/avocado_framework-92.0-py3.9.egg/avocado/core/test.py:793
2022-01-19 18:16:51,025 stacktrace       L0045 ERROR| Traceback (most recent call last):
2022-01-19 18:16:51,025 stacktrace       L0045 ERROR|   File "/home/avocado-fvt-wrapper/tests/avocado-misc-tests/fs/xfstests.py", line 382, in test
2022-01-19 18:16:51,025 stacktrace       L0045 ERROR|     self.fail('One or more tests failed. Please check the logs.')
2022-01-19 18:16:51,025 stacktrace       L0045 ERROR|   File "/usr/local/lib/python3.9/site-packages/avocado_framework-92.0-py3.9.egg/avocado/core/test.py", line 979, in fail
2022-01-19 18:16:51,025 stacktrace       L0045 ERROR|     raise exceptions.TestFail(message)
2022-01-19 18:16:51,025 stacktrace       L0045 ERROR| avocado.core.exceptions.TestFail: One or more tests failed. Please check the logs.
2022-01-19 18:16:51,025 stacktrace       L0046 ERROR| 
2022-01-19 18:16:51,025 test             L0798 DEBUG| Local variables:
2022-01-19 18:16:51,061 test             L0801 DEBUG|  -> self <class 'xfstests.Xfstests'>: 1-xfstests.py:Xfstests.test
2022-01-19 18:16:51,061 test             L0801 DEBUG|  -> failures <class 'bool'>: True
2022-01-19 18:16:51,061 test             L0801 DEBUG|  -> args <class 'str'>: 
2022-01-19 18:16:51,061 test             L0801 DEBUG|  -> cmd <class 'str'>: ./check  -g auto
2022-01-19 18:16:51,061 test             L0801 DEBUG|  -> result <class 'avocado.utils.process.CmdResult'>: command: './check  -g auto'
exit_status: 1
duration: 7138.883506953949
interrupted: False
pid: 3641292
encoding: 'UTF-8'



is this something different? I don't see test Passing. if this is something not relevant or needs a separate PR to fix then  it is OK for me with the changes .
Just Suggestion : adding the snippet of avocado run will give more clarity along with job.log file.

Reply:
@Naresh-ibm for failed tests need to check with dev team, accordingly will take action. This PR was to make xfstest run on rhel 9.0 which was giving below error previously.

05:57:30 INFO    : Running Host Tests Suite distro_regression_xfstests
05:57:30 INFO    : Running: /usr/local/bin/avocado run --test-runner runner avocado-misc-tests/fs/xfstests.py --force-job-id 922d595d7cd5b9c5b964f16ec072e20a13bff93f  --job-results-dir /home/disha/avocado-fvt-wrapper/results
JOB ID     : 922d595d7cd5b9c5b964f16ec072e20a13bff93f
JOB LOG    : /home/disha/avocado-fvt-wrapper/results/job-2021-12-18T05.57-922d595/job.log
 (1/1) avocado-misc-tests/fs/xfstests.py:Xfstests.test:  CANCEL: Fail to install dump required for this test. (1.40 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/disha/avocado-fvt-wrapper/results/job-2021-12-18T05.57-922d595/results.html
JOB TIME   : 15.00 s

--- avocado output after fixing test case ---
[root@ltcden10-lp1 fs]# cat nohup.out 
JOB ID     : 85315632aebc3abdb198b85d8bb1c2826cf5b1e2
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-01-19T16.17-8531563/job.log
 (1/1) xfstests.py:Xfstests.test:  FAIL: One or more tests failed. Please check the logs. (7157.03 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-01-19T16.17-8531563/results.html
JOB TIME   : 7167.48 s

Reply:
@disgoel ok, NP. Thanks for updates.